apps/test/unit/templates/manageStudents/ManageStudentsFamilyNameCellTest.js

```diff
@@ -11,81 +12,45 @@ const DEFAULT_PROPS = {
 };
 
 describe('ManageStudentFamilyNameCell', () => {
-  let editStudent;
-
-  beforeEach(() => {
-    editStudent = sinon.spy();
-  });
-
   it('renders the family name set for the student, when not editing', () => {
-    const wrapper = shallow(
-      <ManageStudentFamilyNameCell
-        {...DEFAULT_PROPS}
-        familyName={'FamName'}
-        editStudent={editStudent}
-      />
+    const wrapper = mount(
+      <Provider store={getStore()}>
+        <ManageStudentFamilyNameCell
+          {...DEFAULT_PROPS}
+          familyName={'FamName'}
+        />
+      </Provider>
     );
     expect(wrapper).to.contain('FamName');
     expect(wrapper.find('input').exists()).to.be.false;
   });
 
   it('renders family name input, when editing', () => {
-    const wrapper = shallow(
-      <ManageStudentFamilyNameCell
-        {...DEFAULT_PROPS}
-        familyName={'FamName'}
-        isEditing={true}
-        editStudent={editStudent}
-      />
+    const wrapper = mount(
+      <Provider store={getStore()}>
+        <ManageStudentFamilyNameCell
+          {...DEFAULT_PROPS}
+          familyName={'FamName'}
+          isEditing={true}
+        />
+      </Provider>
     );
     expect(wrapper.find('input').exists());
   });
 
   it('renders disabled family name input when inputDisabled is true', () => {
-    const wrapper = shallow(
-      <ManageStudentFamilyNameCell
-        {...DEFAULT_PROPS}
-        familyName={'FamName'}
-        isEditing={true}
-        inputDisabled={true}
-        editStudent={editStudent}
-      />
+    const wrapper = mount(
+      <Provider store={getStore()}>
+        <ManageStudentFamilyNameCell
+          {...DEFAULT_PROPS}
+          familyName={'FamName'}
+          isEditing={true}
+          inputDisabled={true}
+        />
+      </Provider>
     );
     const input = wrapper.find('input');
     expect(input.exists());
     expect(input.prop('disabled')).to.be.true;
   });
```

Comment:
Even though we don't pass the redux `editStudent` method in as a property here anymore, it still seems useful to test that changing the input text actually changes the family name in our redux store. Surely there's a way to do that here?

Reply:
I've been reading a lot about testing state for functional components, and I keep finding advice to check that taking an action results in some change to the page (instead of testing state). Would love to have your advice on this because I keep butting my head against this when I'm converting functional components.

Reply:
I think that advice may be good in a lot of cases, but it seems like in this case the key function of the component is that it changes the state so that other components (not being tested here) will show the updated name when we're done inputting new text. Verifying the state change happened would then be what we want to look for in a unit test.

Reply:
As I understand it, we usually use integration tests for exactly that reason in these cases- to avoid testing state in functional components. I'd imagine this kind of test would be exactly what Ryan's would cover if we un-reverted it. That said I'm still trying to make sure redux is working as expected in this conversion anyway (which is why I initially wanted to add UI tests in this pr🫠)

Reply:
Update: I have fixed and merged [Ryan's UI pr](https://github.com/code-dot-org/code-dot-org/pull/53767) and merged those changes into this pr to ensure it is working properly